File: Assets/InsightARWorld/InsightARExporter/Utility/BuiltinLuaScripts/Landmark/BackButton.lua


local BackButton = {}


function BackButton:New(game_object)
    if self ~= BackButton then
        return nil, "BackButton:New is nil"
    end
    local newinstance = setmetatable({}, {__metatable = {}, __index = BackButton})
    newinstance.transform = game_object.transform
    transform = game_object.transform
    newinstance.gameObject = game_object
    gameObject = game_object
    return newinstance
end

function BackButton:Awake()
	
	
end

function BackButton:Start()
	
end

function BackButton:OnPointerUp()
	local _name = self.gameObject.name;
	if _name == "backbutton" then
		g_LandmarkManager:Close();
		Fw_Event_CloseARScene();
	end
end

return BackButton

File: Assets/InsightARWorld/InsightARExporter/Utility/BuiltinJSScripts/Landmark/BackButton.js
function BackButton(gameObject)
{
    this.transform = gameObject.transform;
    this.gameObject = gameObject;
}
//Write prototype function here
BackButton.prototype = Object.assign(Object.create(Object.prototype), {
    // Start is called before the first frame update
    Start: function(){
        
    },
    Awake: function(){

    },
    // Update is called once per frame
    Update: function()
    {
        
    },
    OnPointerUp: function()
	{
        var _name = this.gameObject.name;
        if (_name == "backbutton") {
            if (g_LandmarkManager)
                g_LandmarkManager.Close();
            Fw_Event_CloseARScene();
        }
    }
});

//Return the script module
BackButton